presto-kafka/src/main/java/io/prestosql/plugin/kafka/KafkaConfig.java

```diff
@@ -153,4 +154,17 @@ public KafkaConfig setMessagesPerSplit(int messagesPerSplit)
         this.messagesPerSplit = messagesPerSplit;
         return this;
     }
+
+    public boolean isTimestampUpperBoundPushDownEnabled()
+    {
+        return timestampUpperBoundPushDownEnabled;
+    }
+
+    @Config("kafka.timestamp-upper-bound-push-down-enabled")
```

Comment:
Yes that could be a way to do it, if it is LogAppendTime, then there it is guaranteed that offset = timestamp. This is a setting on the topic which can be retrieved via the Kafka AdminClient. And small example project on how to retrieve `message.timestamp.type` from a topic can be found here: <https://github.com/gschmutz/various-kafka-examples/tree/master/kafka-adminclient-test>. Here is the relevant code:

```
    String topic = "test-topic";
    Properties config = new Properties();
    config.put(AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG, "dataplatform:9092,dataplatform:9093,dataplatform:9094");
    AdminClient admin = KafkaAdminClient.create(config);

    ConfigResource topicResource = new ConfigResource(ConfigResource.Type.TOPIC, topic);

    System.out.println("Getting topic "+topic+" configuration");
    DescribeConfigsResult describeResult = admin.describeConfigs(Collections.singleton(topicResource));
    Map<ConfigResource, Config> topicConfig = describeResult.all().get();
    Config c = topicConfig.get(topicResource);
    System.out.println(c.get("message.timestamp.type").value());
```

Reply:
Thank you for providing the code sample.  This would allow us to get rid of the property altogether and determine on a topic-by-topic case whether it's safe to push down the upper bound of a `_timestamp` range.

Reply:
@aalbu@gschmutz Thank you for your suggestion. I think decided by "message.timestamp.type" is a nice option, but the property is also needed, because even for CreateTime case(default case), pushing down the upper bound of  timestamp is also a normal operation. So I suggest keep property but default not pushing  unless configured.

Reply:
The problem with `CreateTime` is that you are not really sure what the Kafka producer is doing (is he overwriting the timestamp or not, is the system time of the producer client exactly the same as the server time ....) and if you allow the upper bound limit then it is just up to the behavior of the producer client if you can get into the issue I have mentioned or not. So not sure if it is event worth having an upper bound for the `CreateTime` type and if the property not only makes it more complex for a user to decide if it is really worth it.

Reply:
The default operation of CreateTime mode is that it will be generated by Producer API. So the mode will work like what LogAppendTime mode does.  The problem with not supporting pushing down the upper bound is that the timestamp filter will be difficult to use for users with CreateTime mode (may very slow for million of rows maybe).For them, they have no option to push down the upper bound. Would we keep the property, but not pushing the upper bound as default unless configured.

Reply:
@gschmutz hi, gschmutz, the default operation of CreateTime mode is that it will be generated by Producer API. So the mode will work like what LogAppendTime mode does as default.  The problem with not supporting pushing down the upper bound is that the timestamp filter will be difficult to use for users with CreateTime mode (may very slow for million of rows maybe).For them, they have no option to push down the upper bound. Would we keep the property, but not pushing the upper bound as default unless configured for createTime mode.

Reply:
@wangli-td hi, I don't agree that by default `CreateTime` works the same as `LogAppendTime`: if the system clock on client and Kafka cluster is not synchronized/not the same (which easily can happen/ I don't talk of timezone, just system clock sync) then even if the client program does not set the timestamp when producing a record to Kafka, the timestamp the producer client will assign might not exactly be the same as on the server (not even mentioning the lag on the network). Of course the difference would not be seconds but there will be a difference to the time used for the offset (for the index). But I also agree that a push down on upper bound would be nice to have, so we efficiently can select a "time bucket" way in the past. So maybe having a property where you can force the push down of the upper bound when topic is set to `CreateTime`and the default not to do it? I think it is always better if a user has to manually "push" for a feature, which might potentially harm the result.